src/main/java/seedu/duke/assets/DoctorList.java

```diff
@@ -127,38 +127,66 @@ public String toString() {
     }
 
     public void findByNric(String[] parameterArray) {
-        this.returnedFinderArray = DoctorFinder.findDoctorByNric(doctors, parameterArray[1]);
-        createArrayOfFoundDoctors();
+        try {
+            this.returnedFinderArray = DoctorFinder.findDoctorByNric(doctors, parameterArray[1]);
+            createArrayOfFoundDoctors();
+        } catch (NullPointerException e) {
+            UI.printParagraph("Doctor doesn't exist please try again!");
+        }
     }
 
     public void findByName(String[] parameterArray) {
-        this.returnedFinderArray = DoctorFinder.findDoctorByName(doctors, parameterArray[1]);
-        createArrayOfFoundDoctors();
+        try {
+            this.returnedFinderArray = DoctorFinder.findDoctorByName(doctors, parameterArray[1]);
+            createArrayOfFoundDoctors();
+        } catch (NullPointerException e) {
+            UI.printParagraph("Doctor doesn't exist please try again!");
```

Comment:
maybe can customize the exception to return specific error

Reply:
Alrighty!